src/jit/importer.cpp

```diff
@@ -7045,19 +7045,17 @@ GenTreePtr                Compiler::impFixupStructReturn(GenTreePtr     call,
     call->gtCall.gtReturnType = call->gtType;
 
     // Get the classification for the struct.
-    SYSTEMV_AMD64_CORINFO_STRUCT_REG_PASSING_DESCRIPTOR structDesc;
-    eeGetSystemVAmd64PassStructInRegisterDescriptor(retClsHnd, &structDesc);
-    if (structDesc.passedInRegisters)
+    GenTreeCall* callPtr = call->AsCall();
+    eeGetSystemVAmd64PassStructInRegisterDescriptor(retClsHnd, &(callPtr->structDesc));
+    if (callPtr->structDesc.passedInRegisters)
     {
-        call->gtCall.SetRegisterReturningStructState(structDesc);
-
-        if (structDesc.eightByteCount <= 1)
+        if (callPtr->structDesc.eightByteCount <= 1)
         {
-            call->gtCall.gtReturnType = getEightByteType(structDesc, 0);
+            callPtr->gtReturnType = getEightByteType(callPtr->structDesc, 0);
         }
         else
         {
-            if (!call->gtCall.CanTailCall() && ((call->gtFlags & GTF_CALL_INLINE_CANDIDATE) == 0))
+            if (!callPtr->CanTailCall() && ((call->gtFlags & GTF_CALL_INLINE_CANDIDATE) == 0))
```

Comment:
Since you are editing this... the second condition could be simplified/replaced with
!callPtr->IsInlineCandidate()

e.g. if (!callPtr->CanTailCall() && !callPtrIsInlineCandidate())